3 - PB/Documentazione interna/Norme di Progetto/Norme di Progetto.typ

```diff
@@ -1971,27 +1971,44 @@ _Conformant to outcomes to ISO/IEC/IEEE 12207:2017 clause 6.4.11_
-- individuazione di limiti e vincoli che influenzano i requisiti, il design e l'architettura;
-- feedback da parte del Proponente in merito al lavoro svolto;
-- evidenza che il sistema o l'elemento di sistema soddisfa i bisogni del Proponente.
+Come risultato della corretta implementazione del processo di Validazione si ottiene:
+- individuazione di limiti e vincoli che influenzano i requisiti, il design e l'architettura del sistema;
+- feedback da parte del Proponente in merito allo stato di avanzamento e qualità del lavoro svolto;
+- evidenza che il sistema, o l'elemento di sistema, soddisfi i requisiti dichiarati;
+- tracciamento di eventuali anomalie riscontrate.
+
 
 === Attività
-==== Preparazione alla validazione
-La validazione viene svolta prima internamente al gruppo, il Programmatore deve fornire un elemento software che soddisfi le indicazioni date dal Progettista e che possieda le qualità minime di accettabilità per passare la revisione del Verificatore come descritto nei paragrafi del processo di implementazione (@processo_implementazione), di integrazione (@processo_integrazione), e di verifica (@processo_verifica). Il sistema si ritiene quindi in uno stato accettabile, anche se incompleto, e pronto per essere mostrato al Proponente.
+==== Validazione interna
+
+Dopo aver apportato avanzamenti al software e a seguito dell'attuazione del processo di verifica (@processo_verifica), devono avvenire delle analisi e test manuali per constatare lo stato di soddisfacimento dei requisiti. Tali analisi individuali devono venire discusse durante i meeting di retrospettiva in quanto fondamentali per la pianificazione delle future attività.
+I test manuali precedentemente menzionati devono avvenire in un ambiente definito con caratteristiche standard, come l'ambiente predefinito su Azure o mediante l'utilizzo di Docker Compose, e prevedono l'esecuzione manuale delle funzionalità implementate atte a soddisfare i requisiti definiti.
+
+
+==== Validazione esterna
+
+Al fine di esaminare lo stato di avanzamento dei lavori, il gruppo deve pianificare dei meeting settimanali con il Proponente, al quale deve venire presentato il sistema in funzione.
+Esso viene definito come MVP (_Minimum Viable Product_) e rappresenta un prodotto incompleto e incline a ricevere modifiche migliorative, con lo scopo di soddisfare i requisiti minimi definiti.
+
+Il gruppo deve agire conseguentemente al feedback del Proponente riguardo l'adempimento dei bisogni espressi, discutendone durante il meeting interno derivante da tale incontro.
+
+Nel momento in cui il gruppo ritenga il MVP completo di tutte le funzionalità desiderate e definite dalla pianificazione (purché siano soddisfatti tutti i requisiti minimi descritti nel documento #adr_v), deve avvenire un meeting esterno conclusivo nel quale il Proponente, a seguito di un'attenta analisi, afferma il totale adempimento dei bisogni richiesti e riconosce nel prodotto proposto il MVP atteso dal capitolato.
```

Comment:
Riformulerei come segue:

"Nel momento in cui il gruppo avrà soddisfatto tutti i requisiti obbligatori definiti nel documento `#adr_v`, e implementato tutte le funzionalità desiderate, dovrà avvenire il meeting esterno di validazione finale del software. Il gruppo presenterà il prodotto sviluppato al Proponente, il quale, a seguito di un'attenta analisi, potrà affermare il totale adempimento dei requisiti obbligatori e delle funzionalità concordate, riconoscendo il prodotto sviluppato come _Minimum Viable Product_ (MVP) atteso dal capitolato."